### PR TITLE
(skills) Type hinting arm

### PR DESCRIPTION
**Pull request by P-ict0**
Tried to improve type hinting for robot_skills/arm to make it more readable

### CODE REVIEW

File: robot_skills/src/robot_skills/arm/arms.py [hunk 1]
```diff
@@ -112,87 +123,87 @@ def occupied_by(self, value):
...
         self._test_die(self.has_force_sensor, 'has_force_sensor=' + str(self.has_force_sensor),
                        "Specify get_arm(..., force_sensor_required=True)")
-        return self._arm.move_down_until_force_sensor_edge_up(timeout=timeout, retract_distance=retract_distance,
+        return self._arm.move_down_until_force_sensor_edge_up(timeout=timeout,
+                                                              retract_distance=retract_distance,
                                                               distance_move_down=distance_move_down)
 
     # salvaged deprecated functionality
-    def send_gripper_goal(self, state, timeout=5.0, gripper_type=None, max_torque=0.1):
+    def send_gripper_goal(self, state: str, timeout: float = 5.0, gripper_type: str | None = None,
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:198]: Use Optional from typing

File: robot_skills/src/robot_skills/arm/arms.py [hunk 2]
```diff
@@ -67,9 +78,9 @@ class PublicArm(object):
     :vartype _available_joint_trajectories: set of str
 
     """
-    def __init__(self, arm, available_gripper_types, default_gripper_type,
-                 has_occupied_by, allow_force_sensor, available_joint_goals,
-                 available_joint_trajectories):
+    def __init__(self, arm: Arm, available_gripper_types: set[str], default_gripper_type: str | None,
+                 has_occupied_by: bool, allow_force_sensor: bool, available_joint_goals: set[str],
+                 available_joint_trajectories: set[str]) -> None:
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:83]: Use Set from typing module

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Apply comments on the entire file.

File: robot_skills/src/robot_skills/arm/arms.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 from __future__ import print_function, annotations
-from typing import Callable
+from typing import Optional, Union, List, Set, Callable
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py]:
```suggestion
from typing import Callable, List, Optional, Set, Union
```

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Keep import alphabetical

File: robot_skills/src/robot_skills/arm/arms.py [hunk 4]
```diff
@@ -112,87 +122,87 @@ def occupied_by(self, value):
...
+    def send_joint_trajectory(self, configuration: str, timeout: float = JOINT_TIMEOUT,
+                              max_joint_vel: Union[int, float, List[int], List[float]] = MAX_JOINT_VEL) -> bool:
         self._test_die(configuration in self._available_joint_trajectories, 'joint-goal ' + configuration,
                        "Specify get_arm(..., required_trajectories=['{}'])".format(configuration))
         return self._arm.send_joint_trajectory(configuration, timeout=timeout,
                                                max_joint_vel=max_joint_vel)
 
-    def send_goal(self, frameStamped, timeout=30, pre_grasp=False, first_joint_pos_only=False,
-                  allowed_touch_objects=None):
+    def send_goal(self, framestamped: FrameStamped, timeout: float = 30.0, pre_grasp: bool = False,
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:152]:
```suggestion
    def send_goal(self, frame_stamped: FrameStamped, timeout: float = 30.0, pre_grasp: bool = False,
```

File: robot_skills/src/robot_skills/arm/arms.py [hunk 5]
```diff
@@ -112,87 +122,87 @@ def occupied_by(self, value):
...
+                  first_joint_pos_only: bool = False, allowed_touch_objects: Optional[List[str]] = None) -> bool:
         if allowed_touch_objects is None:
             allowed_touch_objects = list()
-        return self._arm.send_goal(frameStamped, timeout, pre_grasp, first_joint_pos_only, allowed_touch_objects)
+        return self._arm.send_goal(framestamped, timeout, pre_grasp, first_joint_pos_only, allowed_touch_objects)
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:156]:
```suggestion
        return self._arm.send_goal(frame_stamped, timeout, pre_grasp, first_joint_pos_only, allowed_touch_objects)
```

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
One final change

**Review by MatthijsBurgh — DISMISSED** [review]

File: robot_skills/src/robot_skills/arm/arms.py [hunk 6]
```diff
@@ -1,25 +1,35 @@
-from __future__ import print_function
+from __future__ import print_function, annotations
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:1]: I think both of these are not needed. `print_function` was needed for python2/3 transition. `annotations` is needed when you want to use a class as type in the class itself.

File: robot_skills/src/robot_skills/arm/arms.py [hunk 7]
```diff
@@ -1,25 +1,35 @@
...
 
 import time
+from typing import Callable, List, Optional, Set, Union
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:4]:
```suggestion
from typing import Callable, List, Optional, Set, Union

import time
```

File: robot_skills/src/robot_skills/arm/arms.py (hunk 6 above)
Comment (P-ict0) [reply robot_skills/src/robot_skills/arm/arms.py]: If we remove annotations we must also change line 80 to `def __init__(self, arm: "Arm", ...`. 
`print_function` we can remove

Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/arm/arms.py]: Oh I missed that one. I will change it.

**Review by MatthijsBurgh — APPROVED** [review]

**Pull request merged**